flood_adapt/object_model/io/unitfulvalue.py

```diff
@@ -100,15 +100,26 @@ def convert(self, new_units: UnitTypesLength) -> float:
             new_conversion = 3.28084
         elif new_units == "inch":
             new_conversion = 1.0 / 0.0254
-        elif self.units == "miles":
+        elif new_units == "miles":
             new_conversion = 1.0 / 1609.344
         else:
             ValueError("Invalid length units")
         return conversion * new_conversion * self.value
 
 
+class UnitfulHeight(UnitfulLength):
```

Comment:
We should make sure this is used everywhere its needed, because currently all heights are just lengths I think, so we should double check every instance of unitfullength and see if it needs to be unitfulheight

Reply:
Added a comment to your pydantic V2 issue #350